### PR TITLE
Prepare for Renovate with reusable workflows

### PR DESCRIPTION
**Pull request by ghostwriter**
|    Q          |   A
|-------------- | ------
| QA            | yes

### Description

Adding Renovate to Laminas, Mezzio repositories. https://gist.github.com/weierophinney/49ca8ae2b916d898dd3d4162b26ff41b

- Configure minimum supported PHP version
- Update composer dependencies
- Update to reusable `Continuous Integration` workflow
- Update to reusable `Automatic Releases` workflow

### CODE REVIEW

**Review by Ocramius — CHANGES_REQUESTED** [review]
CI to be fixed before considering a merge

**Comment by boesing** [comment]
How to add services to the CI pipeline?
Same as in the other components which do need docker services like memcached, redis, and stuff, this is not mergable.
If we do not have a solution for this right now, I'll keep this open until we have

File: .github/workflows/continuous-integration.yml [hunk 1]
```diff
@@ -4,36 +4,8 @@ on:
...
-      - name: Gather CI configuration
-        id: matrix
-        uses: laminas/laminas-ci-matrix-action@v1
-
-  qa:
-    name: QA Checks
-    needs: [matrix]
-    runs-on: ${{ matrix.operatingSystem }}
-    services:
-      memcached:
```
Comment (boesing) [inline .github/workflows/continuous-integration.yml]: must be somehow restored or tests will always fail

**Review by boesing — CHANGES_REQUESTED** [review]

**Comment by Ocramius** [comment]
For having services again, we need to go back, and avoid the reusable workflow here.

**Review by Ocramius — CHANGES_REQUESTED** [review]
Hmm, `8.1` lowest and `8.1` latest seem to fail :thinking:

**Comment by ghostwriter** [comment]
`TypeError` thrown, possibly false positives.
```
TypeError: Memcached::setMulti(): Argument #2 ($expiration) must be of type int, int given
```

**Review by boesing — APPROVED** [review]
Merging for now, tests are also failing in target branch and this looks good from what I can see.

**Pull request merged**